### PR TITLE
web/satellite Generic List component

### PR DESCRIPTION
**Pull request by Barterio**
What: 
    Implemented Generic List component to generalize and combine all list logic to single component
Why:

Please describe the tests:
 - Test 1:
 - Test 2:
 
Please describe the performance impact:

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?
 - [ ] Do the database access patterns make sense?

### CODE REVIEW

File: web/satellite/src/components/test/Item.vue [hunk 1]
```diff
@@ -0,0 +1,26 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+<template>
+    <div>
+            <td>{{data.a }}</td>
```
Comment (crawter) [inline web/satellite/src/components/test/Item.vue]: fix indentation

File: web/satellite/src/components/common/ListComponent.vue [hunk 2]
```diff
@@ -0,0 +1,35 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+<template>
+    <div>
```
Comment (crawter) [inline web/satellite/src/components/common/ListComponent.vue]: remove

File: web/satellite/src/components/common/ListComponent.vue [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        <div v-for="item in dataSet">
+            <component :is="itemComponent" :data="item" v-on:click.native="onItemClick(item)"></component>
```
Comment (crawter) [inline web/satellite/src/components/common/ListComponent.vue]: v-on == @

File: web/satellite/src/components/common/ListComponent.vue (hunk 2 above)
Comment (Barterio) [reply web/satellite/src/components/common/ListComponent.vue]: Repeatable div must be wrapped by div. 
Leaving repeatable div will cause error during render

Comment (NikolaiYurchenko) [reply web/satellite/src/components/common/ListComponent.vue]: Nonsense, you just must give v-for div key attribute

Comment (Barterio) [reply web/satellite/src/components/common/ListComponent.vue]: No it is not. You must have one root element.
Vue interprets the root element having a v-for loop on it as producing multiple root level elements.

File: web/satellite/src/components/common/List.vue [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+            @click.native="onItemClick(item)"
+            v-bind:class="[item.isSelected ? 'selected' : '']"/>
+    </div>
+</template>
+
+<script lang="ts">
+    import { Component, Prop, Vue } from 'vue-property-decorator';
+
+    @Component({
+        components: {},
```
Comment (crawter) [inline web/satellite/src/components/common/List.vue]: why do we have this empty decorator here?

File: web/satellite/src/components/common/List.vue [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    })
+    export default class List<T> extends Vue {
```
Comment (crawter) [inline web/satellite/src/components/common/List.vue]: is there any need for generic class here?

File: web/satellite/src/components/common/List.vue (hunk 4 above)
Comment (Barterio) [reply web/satellite/src/components/common/List.vue]: Good point

**Pull request merged**